go/http/gen_sign_str.go

```diff
@@ -86,8 +85,15 @@ func GetCanonicalRequest(req *http.Request, supportHeaders map[string]struct{})
 // GetMsgToSign generate the msg bytes from canonicalRequest to sign
 func GetMsgToSign(req *http.Request) []byte {
 	headers := initSupportHeaders()
```

Comment:
can we not do `initSupportHeaders` here as it is stateless, let us move it into `GetCanonicalRequest`  so that `GetCanonicalRequest` only need one parameter.